source/bringauto/modules/StatusAggregator.cpp

```diff
@@ -27,7 +27,10 @@ StatusAggregator::aggregateStatus(structures::StatusAggregatorDeviceState &devic
 								  const unsigned int &device_type) {
 	auto &currStatus = deviceState.getStatus();
 	Buffer aggregatedStatusBuff {};
-	module_->aggregateStatus(aggregatedStatusBuff, currStatus, status, device_type);
+	if (module_->aggregateStatus(aggregatedStatusBuff, currStatus, status, device_type) != OK) {
+		log::logWarning("Error occurred while aggregating status, returning empty buffer");
+		aggregatedStatusBuff = module_->constructBuffer();
```

Comment:
What if the aggregatedStatusBuff is already allocated?
Ha, then the Buffer is deallocated and replaced by a new buffer!

But when the constructBUffer is needed here?

Reply:
after discussion, it was decided that module functions will construct an empty buffer if they fail